app/src/main/java/ru/prometeydev/movie/model/MoviesRepository.kt

```diff
@@ -15,11 +15,11 @@ class MoviesRepository(network: MoviesApiProvider): BaseUseCases() {
 
     private lateinit var baseImageUrl: String
 
-    suspend fun loadMovies(): List<Movie> = execute {
+    suspend fun loadMovies(page: Int): List<Movie> = execute {
         baseImageUrl = getImageUrl()
 
         val genresMap = loadGenres()
```

Comment:
Why to load genres all the time?

Reply:
Is it possible to transfer genres loading to the constructor?